numba_cuda/numba/cuda/__init__.py

```diff
@@ -3,23 +3,21 @@
 from numba.core import config
 from .utils import _readenv
 
-# Enable pynvjitlink if the environment variables NUMBA_CUDA_ENABLE_PYNVJITLINK
-# or CUDA_ENABLE_PYNVJITLINK are set, or if the pynvjitlink module is found. If
-# explicitly disabled, do not use pynvjitlink, even if present in the env.
-_pynvjitlink_enabled_in_env = _readenv(
-    "NUMBA_CUDA_ENABLE_PYNVJITLINK", bool, None
-)
-_pynvjitlink_enabled_in_cfg = getattr(config, "CUDA_ENABLE_PYNVJITLINK", None)
-
-if _pynvjitlink_enabled_in_env is not None:
-    ENABLE_PYNVJITLINK = _pynvjitlink_enabled_in_env
-elif _pynvjitlink_enabled_in_cfg is not None:
-    ENABLE_PYNVJITLINK = _pynvjitlink_enabled_in_cfg
-else:
-    ENABLE_PYNVJITLINK = importlib.util.find_spec("pynvjitlink") is not None
-
+# Enable pynvjitlink based on the following precedence:
+# 1. Config setting "CUDA_ENABLE_PYNVJITLINK" (highest priority)
+# 2. Environment variable "NUMBA_CUDA_ENABLE_PYNVJITLINK"
+# 3. Auto-detection of pynvjitlink module (lowest priority)
 if not hasattr(config, "CUDA_ENABLE_PYNVJITLINK"):
```

Comment:
Is there ever a situation where `config.CUDA_ENABLE_PYNVJITLINK` is set to `None` that we need to then check the other options?

Maybe change this to `if getattr(config, "CUDA_ENABLE_PYNVJITLINK", None) is None`?

Reply:
```suggestion
if getattr(config, "CUDA_ENABLE_PYNVJITLINK", None) is None:
```